helpers/auth/authentication.py

```diff
@@ -129,15 +125,25 @@ def wrapper(*args, **kwargs):
                                 to perform this action", status, expected_args)
                         else:
                             handle_http_error(message, status, expected_args)
-
+                    self.save_user()
+                    try:
+                        user = User.query.filter_by(email=email).first()
+                    except Exception:
+                        raise GraphQLError("The database cannot be reached")
                     for value in response['values']:
                         if user.email == value["email"]:
                             if value['location']:
                                 user.location = value['location']['name']
                                 user.save()
+                                add_location(
+                                    value['location']['name']
+                                )  # add location name if not in database
                             else:
                                 user.location = "Nairobi"
                                 user.save()
+                                add_location(
+                                    "Nairobi"
```

Comment:
Good work @verenceLola . However, I didn't get the reason behind defaulting the location to Nairobi yet your PR should aim at saving the user's exact location.

Reply:
Still, if the location of the user cannot be fetched from  the Andela API, it's defaulted to `Nairobi` as before. @TeamoreA

Reply:
Alright👍🏼

Reply:
@verenceLola take this opportunity to refactor the duplication in the `if..else` statement. Remove the `else`. Can we also make the `add_location` function name more descriptive? Because it does not only add the location, it also performs a check. Otherwise well done!